src/jit/lsra.cpp

```diff
@@ -5923,7 +5974,7 @@ void LinearScan::allocateRegisters()
     else
 #endif // DEBUG
     {
-        freeRegisters(regsToFree | delayRegsToFree);
+        freeRegisters(regsToFree);
```

Comment:
so now we only free the current set, why?
What if we had non-zero delayRegsToFree?

Reply:
This situation is similar to the one above, but the assert above was added after a bug was encountered, and it didn't occur to me to add the same here; I've done so now.